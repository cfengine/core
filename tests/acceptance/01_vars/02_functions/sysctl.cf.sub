# you can run this test directly with

# CFENGINE_TEST_OVERRIDE_PROCDIR=`pwd`/01_vars/02_functions/proc ./testall 01_vars/02_functions/sysctl.cf.sub

body common control
{
    inputs => { "../../default.cf.sub" };
    bundlesequence => { default($(this.promise_filename)) };
    version => "1.0";
}

bundle agent test
{
  vars:
      "sysctl" data => data_sysctlvalues();
      "tested" slist => { "net.ipv4.tcp_mem", "net.unix.max_dgram_qlen" };
      "values[$(tested)]" string => sysctlvalue($(tested));
}

bundle agent check
{
  vars:
      "testname" string => regex_replace($(this.promise_filename), "\\.sub$", "", "");

  methods:
      "check"  usebundle => dcs_check_state(test,
                                           "$(this.promise_filename).expected.json",
                                           $(testname));

    test_debug::
      "check"  usebundle => dcs_check_state(test,
                                           "$(this.promise_filename).expected.json",
                                           $(this.promise_filename));
}
