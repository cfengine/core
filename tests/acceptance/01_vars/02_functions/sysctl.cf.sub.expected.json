{
  "sysctl": {
    "net.ipv4.tcp_mem": "383133\t510845\t766266",
    "net.unix.max_dgram_qlen": "512"
  },
  "tested": [
    "net.ipv4.tcp_mem",
    "net.unix.max_dgram_qlen"
  ],
  "values[net.ipv4.tcp_mem]": "383133\t510845\t766266",
  "values[net.unix.max_dgram_qlen]": "512"
}
